utils/utils.ts

```diff
@@ -76,10 +76,17 @@ export function parseDsn(dsn: string): DsnResult {
   const [protocol, strippedUrl] = dsn.match(/(?:(?!:\/\/).)+/g) ?? ["", ""];
   const url = new URL(`http:${strippedUrl}`);
 
+  let password = url.password;
+  // Special characters in the password may be url-encoded by URL(), such as = 
+  try {
+    password = decodeURIComponent(password);
+  } catch (error) {
```

Comment:
A `ConnectionParamsError` must be thrown here if the password encoding fails

Reply:
I mean is that even if the decoding fails, the password is not decoded, which is the current behavior. If an error is to be thrown, I am not sure if there will be an abnormality after the change that the current version can run normally. 

In fact, creating a new URL object is successful, and then url-decoding the password of the url will not fail?

Reply:
I found one that failed, probably shouldn't throw an error

```sh
> let u = new URL('mysql://root:Mtx%3@loalhost:9999/txdb')
undefined
> u
URL {
  href: "mysql://root:Mtx%3@loalhost:9999/txdb",
  origin: "null",
  protocol: "mysql:",
  username: "root",
  password: "Mtx%3",
  host: "loalhost:9999",
  hostname: "loalhost",
  port: "9999",
  pathname: "/txdb",
  hash: "",
  search: ""
}
> decodeURIComponent(u.password)
Uncaught URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at <anonymous>:2:1
> let password = u.password;
undefined
> password
"Mtx%3"
> try {
    password = decodeURIComponent(password);
  } catch (error) {}
undefined
> password
"Mtx%3"

```

Reply:
To be honest I'm not sure what to do regarding this. The most "correct" to me seems to ask users to encode the password beforehand, and not leave it up to us to check whether it's encoded or not, but I can see both cases being equally as valid

You know what? Just add a warning similar to https://github.com/denodrivers/postgres/blob/235d57568bd3a49c353a988918d06077197e9044/connection/connection.ts#L819 indicating the user when the decoding fails and the driver will take the raw password

Reply:
ok, thanks, I added a console.error

Reply:
Let's leave it at
```
URL-encoded password decode failed
Defaulting to take the URL-encoded password
```

No other comments, please rebase over latest main because of a faulty commit

Reply:
I already rebase over latest main.